src/org/zaproxy/zap/extension/formHandler/ExtensionFormHandler.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.formHandler;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import java.util.List;
+
+import org.apache.log4j.Logger;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+
+import org.zaproxy.zap.extension.spider.ExtensionSpider;
+
+public class ExtensionFormHandler extends ExtensionAdaptor {
+
+	public static final String NAME = "ExtensionFormHandler";
+
+	protected static final String PREFIX = "formHandler";
+
+	private FormHandlerParam param;
+
+	private OptionsFormHandlerPanel optionsFormHandlerPanel = null;
+
+    private static final Logger LOGGER = Logger.getLogger(ExtensionFormHandler.class);
+
+    public ExtensionFormHandler() {
+        super(NAME);
+    }
+
+	@Override
+	public void hook(ExtensionHook extensionHook) {
+	    super.hook(extensionHook);
+
+	    extensionHook.addOptionsParamSet(getParam());
+	    ExtensionSpider extension = (ExtensionSpider) Control.getSingleton().getExtensionLoader()
+				.getExtension(ExtensionSpider.NAME);
+	    if (extension != null){
+	    	extension.setValueGenerator(new FormHandlerValueGenerator(getParam()));
+	    }
+
+	    if (getView() != null) {
+	    	extensionHook.getHookView().addOptionPanel(getOptionsFormHandlerPanel());
+	    }
+	}
+
+	@Override
+	public boolean canUnload() {
+		// The extension can be dynamically unloaded, all resources used/added can be freed/removed from core.
+		return true;
+	}
+
+	@Override
+	public void unload() {
+		super.unload();

```

Comment:
It should set the default value generator to the `ExtensionSpider`, otherwise the core/spider would still reference the add-on after uninstalling.